tools/flaky-test-reporter/result.go

```
@@ -0,0 +1,157 @@
+}
+
+func getFlakyRate(testStats map[string]TestStat) (float32, error) {
+	flakyCount := 0
+	totalCount := 0
+	for _, ts := range testStats {
+		totalCount++
+		if ts.isFlaky() {
+			flakyCount++
+		}
+	}
+	if 0 == totalCount {
+		return 0.0, nil
+	}
+	return float32(flakyCount)/float32(totalCount), nil
+}
+
+// createArtifactForRepo marshals RepoData into json format and stores it in a json file,
+// under local artifacts directory
+func createArtifactForRepo(rd *RepoData) error {
+	outFilePath := path.Join(prow.GetLocalArtifactsDir(), rd.Config.Repo + ".json")
+	contents, err := json.Marshal(*rd)
+	if nil != err {
+		return err
+	}
+	return ioutil.WriteFile(outFilePath, contents, 0644)
+}
+
+// addSuiteToRepoData adds all testCase from suite into RepoData
+func addSuiteToRepoData(suite *junit.TestSuite, buildID int, rd *RepoData) {
+	if nil == rd.TestStats {
+		rd.TestStats = make(map[string]*TestStat)
+	}
+	for _, testCase := range suite.TestCases {
+		testFullName := fmt.Sprintf("%s.%s", suite.Name, testCase.Name)
+		if _, ok := rd.TestStats[testFullName]; !ok {
+			rd.TestStats[testFullName] = &TestStat{TestName: testFullName}
+		}
+		switch testCase.GetTestStatus() {
+			case junit.Passed:
+				rd.TestStats[testFullName].Passed = append(rd.TestStats[testFullName].Passed, buildID)
+			case junit.Skipped:
+				rd.TestStats[testFullName].Skipped = append(rd.TestStats[testFullName].Skipped, buildID)
+			case junit.Failed:
+				rd.TestStats[testFullName].Failed = append(rd.TestStats[testFullName].Failed, buildID)
+		}
+	}
+}
+
+// getCombinedResultsForBuild gets all junit results from a build,
+// and converts each one into a junit TestSuites struct
+func getCombinedResultsForBuild(build *prow.Build) []*junit.TestSuites {
+	var allSuites []*junit.TestSuites
+	for _, artifact := range build.GetArtifacts() {
+		relPath, _ := filepath.Rel(build.StoragePath, artifact)
+		contents, err := build.ReadFile(relPath)
+		if nil != err {
+			continue
```

Wouldn't you be silently ignoring problems here?

---

updated to return error if failed parsing junit result